lib/gui/app/modules/update-lock.js

```diff
@@ -0,0 +1,184 @@
+      debug('inactive')
+      this.emit('inactive')
+    })
+  }
+
+  /**
+   * @summary Inactivity event handler, releases the resin update lock on inactivity
+   * @private
+   * @example
+   * this.on('inactive', onInactive)
+   */
+  static onInactive () {
+    if (process.env.ELECTRON_RESIN_UPDATE_LOCK) {
+      UpdateLock.check((checkError, isLocked) => {
+        debug('inactive-check', Boolean(checkError))
+        if (checkError) {
+          analytics.logException(checkError)
+        }
+        if (isLocked) {
+          UpdateLock.release((error) => {
+            debug('inactive-release', Boolean(error))
+            if (error) {
+              analytics.logException(error)
+            }
+          })
+        }
+      })
+    }
+  }
+
+  /**
+   * @summary Acquire the update lock
+   * @private
+   * @param {Function} callback - callback(error)
+   * @example
+   * UpdateLock.acquire((error) => {
+   *   // ...
+   * })
+   */
+  static acquire (callback) {
+    debug('lock')
+    if (process.env.ELECTRON_RESIN_UPDATE_LOCK) {
+      electron.ipcRenderer.once('resin-update-lock', (event, error) => {
+        callback(error)
+      })
+      electron.ipcRenderer.send('resin-update-lock', 'lock')
+    } else {
+      callback(new Error('Update lock disabled'))
+    }
+  }
+
+  /**
+   * @summary Release the update lock
+   * @private
+   * @param {Function} callback - callback(error)
+   * @example
+   * UpdateLock.release((error) => {
+   *   // ...
+   * })
+   */
+  static release (callback) {
+    debug('unlock')
+    if (process.env.ELECTRON_RESIN_UPDATE_LOCK) {
+      electron.ipcRenderer.once('resin-update-lock', (event, error) => {
+        callback(error)
+      })
+      electron.ipcRenderer.send('resin-update-lock', 'unlock')
+    } else {
+      callback(new Error('Update lock disabled'))
+    }
+  }
+
+  /**
+   * @summary Check the state of the update lock
+   * @private
+   * @param {Function} callback - callback(error, isLocked)
+   * @example
+   * UpdateLock.check((error, isLocked) => {
+   *   if (isLocked) {
+   *     // ...
+   *   }
+   * })
+   */
+  static check (callback) {
+    debug('check')
+    if (process.env.ELECTRON_RESIN_UPDATE_LOCK) {
+      electron.ipcRenderer.once('resin-update-lock', (event, error, isLocked) => {
+        callback(error, isLocked)
+      })
+      electron.ipcRenderer.send('resin-update-lock', 'check')
+    } else {
+      callback(new Error('Update lock disabled'))
+    }
+  }
+
+  /**
+   * @summary Extend the lock timer
+   * @example
+   * updateLock.extend()
+   */
+  extend () {
+    debug('extend')
+    this.lockTimer.signal()
+
+    // When extending, check that we have the lock,
+    // and acquire it, if not
+    if (process.env.ELECTRON_RESIN_UPDATE_LOCK) {
+      UpdateLock.check((checkError, isLocked) => {
+        if (checkError) {
+          analytics.logException(checkError)
+        }
+        if (!isLocked) {
```

Comment:
What will be the error of `isLocked` if `checkError` is not `null`?

Reply:
`isLocked` will be undefined in that case. Reasoning behind the mere `!isLocked` is that we'll still want to try to get the lock in that case